pandas/tests/window/test_ewm.py

```diff
@@ -181,3 +181,14 @@ def test_ewma_times_adjust_false_raises():
         Series(range(1)).ewm(
             0.1, adjust=False, times=date_range("2000", freq="D", periods=1)
         )
+
+
+@pytest.mark.parametrize("func", ["mean", "std", "var"])
+@pytest.mark.parametrize("dtype", [np.float32, np.float16, np.float64, float, "float"])
+def test_float_dtype_ewma(dtype, func):
+    # GH#42452
+    df = DataFrame(np.random.rand(20, 3), dtype=dtype)
```

Comment:
Could you use constant data, build a `expected = DataFrame(...)` and use `tm.assert_frame_equal(result, expected)`?